File: archive/cdk-ts/Pr5934/lib/environment-config.ts
export interface EnvironmentConfig {
  vpcCidr: string;
  maxAzs: number;
  rdsInstanceClass: string;
  rdsBackupRetention: number;
  rdsMultiAz: boolean;
  lambdaMemorySize: number;
  logRetention: number;
  s3Versioning: boolean;
  dynamodbBillingMode: 'PAY_PER_REQUEST' | 'PROVISIONED';
  dynamodbReadCapacity?: number;
  dynamodbWriteCapacity?: number;
}

const configs: Record<string, EnvironmentConfig> = {
  dev: {
    vpcCidr: '10.0.0.0/16',
    maxAzs: 2,
    rdsInstanceClass: 'db.t3.micro',
    rdsBackupRetention: 7,
    rdsMultiAz: false,
    lambdaMemorySize: 512,
    logRetention: 7,
    s3Versioning: false,
    dynamodbBillingMode: 'PAY_PER_REQUEST',
  },
  staging: {
    vpcCidr: '10.1.0.0/16',
    maxAzs: 2,
    rdsInstanceClass: 'db.t3.small',
    rdsBackupRetention: 14,
    rdsMultiAz: false,
    lambdaMemorySize: 1024,
    logRetention: 30,
    s3Versioning: true,
    dynamodbBillingMode: 'PAY_PER_REQUEST',
  },
  prod: {
    vpcCidr: '10.2.0.0/16',
    maxAzs: 2,
    rdsInstanceClass: 'db.r5.large',
    rdsBackupRetention: 30,
    rdsMultiAz: true,
    lambdaMemorySize: 2048,
    logRetention: 90,
    s3Versioning: true,
    dynamodbBillingMode: 'PROVISIONED',
    dynamodbReadCapacity: 5,
    dynamodbWriteCapacity: 5,
  },
};

// FIX 5: Environment validation with proper error message
export function getEnvironmentConfig(env: string): EnvironmentConfig {
  const validEnvironments = ['dev', 'staging', 'prod'];
  if (!validEnvironments.includes(env)) {
    throw new Error(
      `Invalid environment: ${env}. Valid values: ${validEnvironments.join(', ')}`
    );
  }
  return configs[env];
}

File: archive/cdktf-py/Pr7824/lib/environment_config.py
"""Environment-specific configuration for multi-environment deployment."""

from typing import Dict, Any


class EnvironmentConfig:
    """Configuration manager for environment-specific settings."""

    # Environment-specific configurations
    CONFIGS: Dict[str, Dict[str, Any]] = {
        "dev": {
            "lambda_memory": 256,
            "lambda_timeout": 30,
            "rds_backup_retention": 1,
            "rds_instance_class": "db.t3.micro",
            "rds_multi_az": False,
            "dynamodb_billing_mode": "PAY_PER_REQUEST",
            "dynamodb_read_capacity": None,
            "dynamodb_write_capacity": None,
            "s3_versioning_enabled": False,
            "cloudwatch_log_retention": 7,
            "api_stage_name": "dev",
        },
        "staging": {
            "lambda_memory": 512,
            "lambda_timeout": 60,
            "rds_backup_retention": 7,
            "rds_instance_class": "db.t3.small",
            "rds_multi_az": False,
            "dynamodb_billing_mode": "PAY_PER_REQUEST",
            "dynamodb_read_capacity": None,
            "dynamodb_write_capacity": None,
            "s3_versioning_enabled": False,
            "cloudwatch_log_retention": 30,
            "api_stage_name": "staging",
        },
        "prod": {
            "lambda_memory": 1024,
            "lambda_timeout": 120,
            "rds_backup_retention": 30,
            "rds_instance_class": "db.t3.medium",
            "rds_multi_az": False,
            "dynamodb_billing_mode": "PROVISIONED",
            "dynamodb_read_capacity": 5,
            "dynamodb_write_capacity": 5,
            "s3_versioning_enabled": True,
            "cloudwatch_log_retention": 90,
            "api_stage_name": "prod",
        },
    }

    @classmethod
    def get_config(cls, environment: str) -> Dict[str, Any]:
        """
        Get configuration for specified environment.

        Args:
            environment: Environment name (dev, staging, prod)

        Returns:
            Configuration dictionary for the environment

        Raises:
            ValueError: If environment is not recognized
        """
        if environment not in cls.CONFIGS:
            raise ValueError(
                f"Unknown environment: {environment}. "
                f"Valid options: {', '.join(cls.CONFIGS.keys())}"
            )
        return cls.CONFIGS[environment]

    @classmethod
    def get_vpc_cidr(cls, environment: str) -> str:
        """
        Get VPC CIDR block for environment.

        Args:
            environment: Environment name (dev, staging, prod)

        Returns:
            CIDR block for the environment
        """
        vpc_cidrs = {
            "dev": "10.0.0.0/16",
            "staging": "10.1.0.0/16",
            "prod": "10.2.0.0/16",
        }
        return vpc_cidrs.get(environment, "10.0.0.0/16")
